_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,867 @@
+
+Add a second Docker registry trigger for the gateway image pushes and click **Save Changes**.  
+
+### Configure the deployment stages
+
+The store pipeline will execute the following stages:
+
+- Deploy the application ConfigMap
+- Deploy the JHipster Registry
+- Deploy the gateway database PostgreSQL
+- Deploy the store clustered database MongoDB
+- Deploy the gateway microservice
+- Deploy the store microservice
+- Enable the gateway service
+- Enable the store service
+
+The Kubernetes configuration created in the folder `jhipster-ci-cd/kubernetes` must be pushed to a GitHub repository, so Spinnaker can access it using the GitHub artifact account created earlier. But before you push, edit `kubernetes/store-k8s/store-deployment.yml` and set the image name with the fully qualified address:
+
+```yml
+- name: store-app
+  image: index.docker.io/indiepopart/store
+```
+Do the same for `kubernetes/gateway-k8s/gateway-deployment.yml`. This is also required for the artifact substitution to work.
+
+Sign in to GitHub and create a public repository `jhipster-k8s`. Follow the instructions to push the existing `kubernetes` folder from your local using the command line.
+
+**IMPORTANT NOTE**: At this point, you will be pushing plain secrets contained in the application and Kubernetes configuration. To avoid this insecure anti-pattern, you can run the JHipster registry locally to encrypt application configuration, and also set up `kubeseal` for Kubernetes secrets encryption. The process is described in a previous post [Kubernetes to the Cloud with Spring Boot and JHipster](/blog/2021/06/01/kubernetes-spring-boot-jhipster).
+
+Go back to Spinnaker UI, and in the pipeline configuration choose **Add Stage** and set the following values:
+
+- Type: **Deploy(Manifest)**
+- Stage Name: **deploy application-configmap**
+- Account: **jhipster-gke-account**
+- Override Namespace: **yes**
+- Namespace: **demo**
+- Manifest Source: **Artifact**
+- Manifest Artifact: **Define a new artifact**
+
+ In the artifact fields, set the following values:
+
+ - Account: **your-github-username**
+ - Content URL: https://api.github.com/repos/your-github-username/jhipster-k8s/contents/registry-k8s/application-configmap.yml
+ - Commit/Branch: **main**
+
+ Click **Save Changes**.
+ Repeat the process for adding stages to deploy `jhipster-registry.yml`, which must depend on the previous manifest. Also repeat for `gateway-postgresql.yml`, `store-mongodb.yml` manifests, which both should depend on the `jhipster-registry` stage.
```

Comment:
I'd add a note here explaining that you have to use `api.github.com` for all these URLs and maybe even mention that it should render a JSON response, not a web page or the raw YAML. For example, https://api.github.com/repos/mraible/jhipster-k8s/contents/registry-k8s/jhipster-registry.yml

Reply:
Added the note.